### ISSUE TITLE
Improved naming/folder sorting

### ISSUE THREAD

**Issue by X39** [Krzmbrzl/ArmaPreprocessorTestCases#1]
Currently,  checking if a case is already covered
eg.
```sqf
#define MACRO(a) \ 
  diag_log str(a)

MACRO(1);
```
--> (proper result not tested yet, may differ)
```sqf

  diag_log 1

\ 
```
requires opening literally every case and checking it.
It would be nice, if some sort of sorting would exist, that could tell them appart before even opening and thus allowing for quick checking if a given case is already implemented in the expected range.

**Comment by Krzmbrzl** [comment]
I fully agree with you.
I just haven't had the time to do so. Also I am not quite sure how such a sorting would be done best. Obviously a separation between all invalid test-cases (test-cases testing invalid preproc-input) and the valid ones. But besides that...?

Do you have an idea how one could approach this?

**Comment by X39** [comment]
well ... one could name them proper for starters 🙈 

for example, the above could be named `backspace_lineescape_whitespace` 🤷‍♂️

**Comment by Krzmbrzl** [comment]
But in order to name every test case properly, precisely and most important: uniquely, the names would have to be so verbose, that I'm afraid it won't be practical to do so (or rather it wouldn't be practical to search for stuff either as reading the filename requires as much time as actually opening the file and having a look at its content).

**Comment by Krzmbrzl** [comment]
I think creating a folder structure is better suited for this case. I'd suggest creating folders that describe mutually exclusive categories each test falls in.

The first layer might be what kind of preprocessing is tested: `include`, `define` , `undef`, `if(n)def` and `mixed`
Then inside there a finer differentiation is being made. As most test-cases currently present in this repo focus on defines, I will demonstrate what I mean for that category.
As the second layer in the define-category one could start by creating folders like `no-arguments`, `one-argument` and `multi-argument`. The third layer then has to be present in all sub-folders representing the second layer. Such a third layer might be `single-line` and `multi-line`.

So all in all the file-structure described so far would look like this:
```
- include
- define
  - no-arguments
    - single-line
    - multi-line
  - one-argument
    - single-line
    - multi-line
  - multi-argument
    - single-line
    - multi-line
- undef
- if(n)def
- mixed
```

Of course this could be continued allowing for a database-like access of the files where one can look up exactly what one is looking for. Once that file-structure is in place one can additionally work with more descriptive filenames but those no longer have to describe the whole file content (as this is done by where the file is placed in the proposed structure) and can thus be shorter and more readable.
What do you think about this system?

**Comment by X39** [comment]
looks fine to me

**Comment by Krzmbrzl** [comment]
Is the current folder structure sufficient?

**Comment by X39** [comment]
i would say, yes 😄

**Comment by Krzmbrzl** [comment]
Results are now finally sorted as well. Closing this now
